Kitodo/src/main/java/org/kitodo/production/forms/createprocess/ProcessDataTab.java

```diff
@@ -192,23 +200,85 @@ public void generateProcessTitleAndTiffHeader() {
         Process process = this.createProcessForm.getMainProcess();
         try {
             StructuralElementViewInterface docTypeView = createProcessForm.getRulesetManagement().getStructuralElementView(
-                docType, createProcessForm.getAcquisitionStage(), createProcessForm.getPriorityList());
+                    docType, createProcessForm.getAcquisitionStage(), createProcessForm.getPriorityList());
             String processTitle = docTypeView.getProcessTitle().orElse("");
             if (processTitle.isEmpty()) {
                 Helper.setErrorMessage("newProcess.titleGeneration.creationRuleNotFound",
                     new Object[] {getDocTypeLabel(docType), process.getRuleset().getTitle() });
             }
-            this.atstsl = ProcessService.generateProcessTitle(this.atstsl, processDetails,
-                processTitle, process);
+
+            String currentTitle = TitleGenerator.getValueOfMetadataID(TitleGenerator.TITLE_DOC_MAIN, processDetails);
+
+            if (StringUtils.isBlank(currentTitle)) {
+                Process parentProcess = createProcessForm.getTitleRecordLinkTab().getTitleRecordProcess();
+                if (Objects.nonNull(parentProcess)) {
+                    currentTitle = getTitleFromMetaXML(parentProcess);
+                } else {
+                    currentTitle = getTitleFromAncestors();
+                }
+            }
+
+            String atstsl = ProcessService.generateProcessTitleAndGetAtstsl(processDetails, processTitle, process,
+                currentTitle);
+
             // document name is generally equal to process title
             this.tiffHeaderDocumentName = process.getTitle();
-            this.tiffHeaderImageDescription = ProcessService.generateTiffHeader(
-                    processDetails, this.atstsl, ServiceManager.getImportService().getTiffDefinition(), this.docType);
+            this.tiffHeaderImageDescription = ProcessService.generateTiffHeader(processDetails, atstsl,
+                ServiceManager.getImportService().getTiffDefinition(), this.docType);
         } catch (ProcessGenerationException e) {
             Helper.setErrorMessage(e.getLocalizedMessage(), logger, e);
         }
         Ajax.update("editForm:processFromTemplateTabView:processDataEditGrid",
-                "editForm:processFromTemplateTabView:processMetadata");
+            "editForm:processFromTemplateTabView:processMetadata");
+    }
+
+    private String getTitleFromMetaXML(Process process) {
+        String xpath = "//kitodo:metadata[@name='" + TitleGenerator.TITLE_DOC_MAIN + "']";
+        try {
+            NodeList nodeList = ServiceManager.getProcessService().getNodeListFromMetadataFile(process, xpath);
+            if (nodeList.getLength() > 0) {
+                return nodeList.item(0).getTextContent();
+            }
+        } catch (IOException e) {
+            logger.error(e.getMessage(), e);
+        }
+        return null;
```

Comment:
The result of this function call is _currently_ only used in `generateProcessTitleAndGetAtstsl`, where a null check is performed. In case of `null`, the value is replaced with an empty String. 

However if a developer uses this method in the future and does _not_ perform a null check, an exception might occur. I think it would be safer to return the empty String here to avoid potential `NullPointerExceptions` if other methods call this function in the future, and check for empty String in  `generateProcessTitleAndGetAtstsl`, instead.